File: arreglo.py
ventas = [[[] for _ in range(3)] for _ in range(12)]
meses = ["Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
         "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"]
departamentos = ["Ropa", "Deportes", "Juguetería"]

def insertar_venta():
    print("\nInsertar una venta:")
    for i, mes in enumerate(meses):
        print(f"{i + 1}. {mes}")
    mes_index = int(input("Selecciona un mes (1-12): ")) - 1
    for i, dept in enumerate(departamentos):
        print(f"{i + 1}. {dept}")
    dep_index = int(input("Selecciona un departamento (1-3): ")) - 1
    monto = input("Ingresa el monto de la venta: ")
    
    ventas[mes_index][dep_index].append(monto)
    print("Venta agregada correctamente.")

def buscar_venta():
    print("\nBuscar una venta:")
    for i, mes in enumerate(meses):
        print(f"{i + 1}. {mes}")
    mes_index = int(input("Selecciona un mes (1-12): ")) - 1
    for i, dept in enumerate(departamentos):
        print(f"{i + 1}. {dept}")
    dep_index = int(input("Selecciona un departamento (1-3): ")) - 1
    if ventas[mes_index][dep_index]:
        print("Ventas encontradas:", ", ".join(ventas[mes_index][dep_index]))
    else:
        print("No hay ventas registradas en esa posición.")

def eliminar_venta():
    print("\nEliminar ventas:")
    for i, mes in enumerate(meses):
        print(f"{i + 1}. {mes}")
    mes_index = int(input("Selecciona un mes (1-12): ")) - 1
    for i, dept in enumerate(departamentos):
        print(f"{i + 1}. {dept}")
    dep_index = int(input("Selecciona un departamento (1-3): ")) - 1
    ventas[mes_index][dep_index] = []
    print("Ventas eliminadas correctamente.")

def mostrar_tabla():
    print("\nTabla de ventas:")
    print("{:<12} {:<12} {:<12} {:<12}".format("", *departamentos))
    for i, mes in enumerate(meses):
        if any(ventas[i][j] for j in range(len(departamentos))):
            fila = []
            for j in range(len(departamentos)):
                if ventas[i][j]:
                    fila.append(", ".join(ventas[i][j]))
                else:
                    fila.append("")
            print("{:<12} {:<12} {:<12} {:<12}".format(mes, fila[0], fila[1], fila[2]))

while True:
    print("\n--- Menú ---")
    print("1. Insertar venta")
    print("2. Buscar venta")
    print("3. Eliminar venta")
    print("4. Mostrar tabla de ventas")
    print("5. Salir")
    
    opcion = input("Selecciona una opción: ")
    
    if opcion == "1":
        insertar_venta()
    elif opcion == "2":
        buscar_venta()
    elif opcion == "3":
        eliminar_venta()
    elif opcion == "4":
        mostrar_tabla()
    elif opcion == "5":
        print("Saliendo del programa...")
        break
    else:
        print("Opción no válida. Intenta de nuevo.")

File: Arreglo.java
import java.util.Scanner;

public class Arreglo {

    static String[][] ventas = new String[12][3];
    static String[] meses = {
        "Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
        "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"
    };
    static String[] departamentos = {"Ropa", "Deportes", "Juguetería"};

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        while (true) {
            System.out.println("\n--- Menú ---");
            System.out.println("1. Insertar venta");
            System.out.println("2. Buscar venta");
            System.out.println("3. Eliminar venta");
            System.out.println("4. Mostrar tabla de ventas");
            System.out.println("5. Salir");
            System.out.print("Selecciona una opción: ");
            int opcion = scanner.nextInt();
            scanner.nextLine(); 

            if (opcion == 1) {
                insertarVenta(scanner);
            } else if (opcion == 2) {
                buscarVenta(scanner);
            } else if (opcion == 3) {
                eliminarVenta(scanner);
            } else if (opcion == 4) {
                mostrarTabla();
            } else if (opcion == 5) {
                System.out.println("¡Adiós!");
                break;
            } else {
                System.out.println("Opción no válida. Intenta de nuevo.");
            }
        }
        scanner.close();
    }

    public static void insertarVenta(Scanner scanner) {
        System.out.println("\nInsertar una venta:");
        
        for (int i = 0; i < meses.length; i++) {
            System.out.println((i + 1) + ". " + meses[i]);
        }
        System.out.print("Selecciona un mes (1-12): ");
        int mesIndex = scanner.nextInt() - 1;
        
        for (int i = 0; i < departamentos.length; i++) {
            System.out.println((i + 1) + ". " + departamentos[i]);
        }
        System.out.print("Selecciona un departamento (1-3): ");
        int depIndex = scanner.nextInt() - 1;
        
        scanner.nextLine(); 
        System.out.print("Ingresa el monto de la venta: ");
        String monto = scanner.nextLine();
        
        ventas[mesIndex][depIndex] = monto;
        System.out.println("Venta agregada correctamente.");
    }

    public static void buscarVenta(Scanner scanner) {
        System.out.println("\nBuscar una venta:");
        
        for (int i = 0; i < meses.length; i++) {
            System.out.println((i + 1) + ". " + meses[i]);
        }
        System.out.print("Selecciona un mes (1-12): ");
        int mesIndex = scanner.nextInt() - 1;
        
        for (int i = 0; i < departamentos.length; i++) {
            System.out.println((i + 1) + ". " + departamentos[i]);
        }
        System.out.print("Selecciona un departamento (1-3): ");
        int depIndex = scanner.nextInt() - 1;
        
        if (!ventas[mesIndex][depIndex].isEmpty()) {
            System.out.println("Venta encontrada: " + ventas[mesIndex][depIndex]);
        } else {
            System.out.println("No hay venta registrada en esa posición.");
        }
    }

    public static void eliminarVenta(Scanner scanner) {
        System.out.println("\nEliminar una venta:");
        
        for (int i = 0; i < meses.length; i++) {
            System.out.println((i + 1) + ". " + meses[i]);
        }
        System.out.print("Selecciona un mes (1-12): ");
        int mesIndex = scanner.nextInt() - 1;
        
        for (int i = 0; i < departamentos.length; i++) {
            System.out.println((i + 1) + ". " + departamentos[i]);
        }
        System.out.print("Selecciona un departamento (1-3): ");
        int depIndex = scanner.nextInt() - 1;
        
        ventas[mesIndex][depIndex] = "";
        System.out.println("Venta eliminada correctamente.");
    }

    public static void mostrarTabla() {
        System.out.println("\nTabla de ventas:");
        System.out.printf("%-12s %-12s %-12s %-12s\n", "", departamentos[0], departamentos[1], departamentos[2]);
        
        for (int i = 0; i < meses.length; i++) {
            System.out.printf("%-12s %-12s %-12s %-12s\n", meses[i], ventas[i][0], ventas[i][1], ventas[i][2]);
        }
    }
}
